### PR TITLE
Add multiple auth token

### PR DESCRIPTION
**Pull request by dtrai2**
Adds the option to use multiple auth token when using the `oauth` authentication method. Additionally, this pull request adds a short SHA to the container images.

## ToDos:
- [x] implement search for correct token
- [x] add test for getter
- [x] fix test pollution in `test_getter.py`
- [x] ~add SHA to CI Pipeline~ (removed short sha from tag again, as it spams the registry)

### CODE REVIEW

File: logprep/util/getter.py [hunk 1]
```diff
@@ -154,15 +168,32 @@ def get_raw(self) -> bytearray:
...
-                resp = session.get(
-                    url=url,
-                    timeout=5,
-                    allow_redirects=True,
-                    headers=self._headers,
-                )
+                if not self._found_valid_token and self._tokens:
+                    resp = self._search_for_valid_token(session, url)
+                else:
+                    resp = self._execute_request(session, url)
```
Comment (ekneg54) [inline logprep/util/getter.py:174]: this confuses me. `_search_for_valid_token` obvisously does more than searching for a valid token. please refactor for cleaner code and better understanding

**Review by ekneg54 — CHANGES_REQUESTED** [review]
some remarks. please have a look

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -175,6 +175,10 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Set outputs
+        id: vars
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
```
[the remark this answers is not in the document]
Comment (dtrai2) [reply .github/workflows/ci.yml]: I removed it again

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -184,3 +188,4 @@ jobs:
             PYTHON_VERSION=${{ matrix.python-version }}
           tags: |
             ghcr.io/fkie-cad/logprep:py${{ matrix.python-version }}-${{ github.head_ref }}
+            ghcr.io/fkie-cad/logprep:py${{ matrix.python-version }}-${{ github.head_ref }}-${{ steps.vars.outputs.sha_short }}
```
[the remark this answers is not in the document]
Comment (dtrai2) [reply .github/workflows/ci.yml]: I removed it again

File: logprep/util/getter.py [hunk 4]
```diff
@@ -129,10 +138,15 @@ def __attrs_post_init__(self):
...
         if os.environ.get("LOGPREP_CONFIG_AUTH_METHOD") == "oauth":
+            self._username = None
+            self._password = None
             if token := os.environ.get("LOGPREP_CONFIG_AUTH_TOKEN"):
-                self._headers.update({"Authorization": f"Bearer {token}"})
-                self._username = None
-                self._password = None
+                self._tokens.append(token)
+            for token_index in count(start=0, step=1):
+                if token := os.environ.get(f"LOGPREP_CONFIG_AUTH_TOKEN_{token_index}"):
```
[the remark this answers is not in the document]
Comment (dtrai2) [reply logprep/util/getter.py]: That doesn't work with the walrus-operator. Now the variable `token` is always a boolean, the evaluation of the expression `environ.get() is None`. So in favor of it being explicit I split it up into two lines.

File: logprep/util/getter.py (hunk 1 above)
Comment (dtrai2) [reply logprep/util/getter.py]: I revised it a bit and added some comments. Please have a look at it again.

**Comment by dtrai2** [comment]
closed in favor of #531

**Pull request closed without merging**